src/braket/circuits/noises.py

```diff
@@ -0,0 +1,252 @@
+from typing import Iterable
+
+import numpy as np
+
+import braket.ir.jaqcd as ir
+from braket.circuits import circuit
+from braket.circuits.instruction import Instruction
+from braket.circuits.noise import Noise, ProbabilityNoise
+from braket.circuits.quantum_operator_helpers import (
+    is_CPTP,
+    verify_quantum_operator_matrix_dimensions,
+)
+from braket.circuits.qubit_set import QubitSet, QubitSetInput
+
+"""
+To add a new noise:
+    1. Implement the class and extend `Noise`
+    2. Add a method with the `@circuit.subroutine(register=True)` decorator. Method name
+       will be added into the `Circuit` class. This method is the default way
+       clients add this noise to a circuit.
+    3. Register the class with the `Noise` class via `Noise.register_noise()`.
+"""
+
+
+class Bit_Flip(ProbabilityNoise):
+    """Bit flip noise channel."""
+
+    def __init__(self, prob: float):
+        super().__init__(
+            prob=prob, qubit_count=1, ascii_symbols=["NB({:.2g})".format(prob)],
```

Comment:
Google Cirq uses the similar convention, but without prefix "N". I think the idea is to avoid super long circuit printout. I will add the display names to the docstring so that users are not confused.

Reply:
@kshitijc I think we also want it to be concise as well. Otherwise the circuit printout would get very long easily. We can use Google's convention but add a prefix "N", which make them slightly more descriptive but are still concise: "NBF" (bit flip), "NPF" (phase flip), "NAD" (amplitude damping), "ND" (depolarizing) and "NK" (kraus). Let me know what you think.